### ISSUE TITLE
Conflict between structs

### ISSUE THREAD

**Issue by padawin** [swaggo/swag#1054]
In a doc comment, I have: 

```
// @Success 200 {object} somepackage.SomeStruct
```

But in my codebase I have 2 packages named `somepackage` which both have a type named `SomeStruct`. When I generate the documentation, `swag` does not know which one to pick and picks either one randomly, making the documentation being sometimes badly generated.

My question would be:

How could I, in the doc comment, specify which is the good package (directory) to be used, without renaming the structs, or packages?

**Comment by josephzidell** [comment]
In your file, only one of them can be imported using that name. That's the one swag uses

**Comment by josephzidell** [comment]
Or try giving it the full path, ala
```go
// @Success 200 {object} full/path/to/somepackage.SomeStruct
```

**Comment by HYY-yu** [comment]
why not import the package in go file:
```
import (
	"encoding/json"
	"net/http"

	_ "full/path/to/somepackage.SomeStruct"
)

```

**Comment by padawin** [comment]
The struct is in the same package as the handler's (where the swagger doc comment is), so I can't import it.

I would like to take the route of giving the full path, but neither of these work:

```
// @Success 200 {object} github.com/organisation/project/full/path/to/somepackage.SomeStruct
// @Success 200 {object} organisation/project/full/path/to/somepackage.SomeStruct
// @Success 200 {object} project/full/path/to/somepackage.SomeStruct
// @Success 200 {object} full/path/to/somepackage.SomeStruct
```

The result is always:

```2021/11/22 11:55:03 ParseComment error in file path/to/package/file.go :can not find schema type: "<any of the above paths>"```

Am I missing something is the format that `full/path/to/somepackage.SomeStruct` should have?

**Comment by padawin** [comment]
Also, using:

```
// @Success 200 {object} SomeStruct
```

(because the package is the same as the current file)

Behaves the same as using `package.SomeStruct`, it conflicts between the two and uses randomly either.

**Comment by HYY-yu** [comment]
I created your example, but did not find your problem.

####  api.go
```
package api

// GetStringByInt example
// @Summary Add a new pet to the store
// @Description get string by ID
// @ID get-string-by-int
// @Accept  json
// @Produce  json
// @Param   some_id      path   int     true  "Some ID"
// @Success 200 {string} string	"ok"
// @Failure 400 {object} api.A "We need ID!!"
// @Router /testapi/get-string-by-int/{some_id} [get]
func GetStringByInt(w http.ResponseWriter, r *http.Request) {
	var pet web.Pet
	if err := json.NewDecoder(r.Body).Decode(&pet); err != nil {
		// write your code
		return
	}

	// write your code
}

type A struct {
	AA int `json:"aa"`
}
```
#### api2.go

```
package api

import (
	"encoding/json"
	"net/http"

	"github.com/swaggo/swag/example/basic/web"
	"github.com/swaggo/swag/example/basic/web/api" // another api package.
)

// GetStringByInt example
// @Summary Add a new pet to the store
// @Description get string by ID
// @ID get-string-by-int2
// @Accept  json
// @Produce  json
// @Param   some_id      path   int     true  "Some ID"
// @Success 200 {string} string	"ok"
// @Failure 400 {object} api.A "We need ID!!"
// @Router /testapi/get-string-by-int2/{some_id} [get]
func GetStringByInt2(w http.ResponseWriter, r *http.Request) {
	_ = api.A{}
	var pet web.Pet
	if err := json.NewDecoder(r.Body).Decode(&pet); err != nil {
		// write your code
		return
	}

	// write your code
}
```

#### swaggo.json

```
        "github.com_swaggo_swag_example_basic_api.A": {
            "type": "object",
            "properties": {
                "aa": {
                    "type": "integer"
                }
            }
        },
        "github.com_swaggo_swag_example_basic_web_api.A": {
            "type": "object",
            "properties": {
                "aaa": {
                    "type": "integer"
                },
                "cc": {
                    "type": "string"
                }
            }
        },
```

I use the code of the master branch.

**Comment by padawin** [comment]
Ok, I could narrow things down. On our project, we are using `swag` 1.6.7. 

This version does not require packages to be imported locally to be present in the swagger doc. I can reproduce the issue in a minimum working example (see attached, also is a tar.gz file ok?).

There are 2 structs `MyStruct` in 2 packages named `mypackage` (one in a `withname` directory, for which `MyStruct` has a `name` field and one in a `withtitle` directory, for which `MyStruct` has a `title` field). The swagger doc says that the endpoint returns `mypackage.MyStruct`, the used one with `withname`, and if you run `swag init`, sometimes the doc files get updated as follow:

```
> git diff
diff --git a/docs/docs.go b/docs/docs.go
index 3f4da35..6263bcc 100644
--- a/docs/docs.go
+++ b/docs/docs.go
@@ -69,7 +69,7 @@ var doc = `{
                 "id": {
                     "type": "string"
                 },
-                "name": {
+                "title": {
                     "type": "string"
                 }
             }
diff --git a/docs/swagger.json b/docs/swagger.json
index 260aab3..c252b29 100644
--- a/docs/swagger.json
+++ b/docs/swagger.json
@@ -49,7 +49,7 @@
                 "id": {
                     "type": "string"
                 },
-                "name": {
+                "title": {
                     "type": "string"
                 }
             }
diff --git a/docs/swagger.yaml b/docs/swagger.yaml
index 4db353a..859fa06 100644
--- a/docs/swagger.yaml
+++ b/docs/swagger.yaml
@@ -3,7 +3,7 @@ definitions:
     properties:
       id:
         type: string
-      name:
+      title:
         type: string
     type: object
 info:
```

Which is the behaviour of this issue.

If I switch to `swag 1.7.4`, I *have* to explicitly have the type imported in my file.

I will see in my project to migrate to `swag 1.7.4`, which would then solve the issue.

I am not sure whether you'd then consider this a bug in `1.6`, so please feel free to close the ticket if it is a case of `won't solve`. 

[myapi.tar.gz](https://github.com/swaggo/swag/files/7580972/myapi.tar.gz)

**Comment by padawin** [comment]
Also, using:

```
// @Success 200 {object} SomeStruct
```

(because the package is the same as the current file)

Behaves the same as using `package.SomeStruct`, it conflicts between the two and uses randomly either.

**Comment by HYY-yu** [comment]
I'm not the maintainer of this project. But I recommend you use the latest version, It will contain fixes to previous versions.

**Comment by ubogdan** [comment]
@padawin We are aware of the bug you are reporting. As far as I know, it was fixed a while ago. 
As you may notice, swag is `go.mod` enabled, and there is no way of applying a patch to that version. 
On the other hand, it should be easy to update your code to work with the latest swag version.

**Comment by sdghchj** [comment]
This is what I have made effort to fix from v1.6.8.
Just use the latest version.

```
//api.go

import (
      somepackage1  aaa/bbb/somepackage
      somepackage2  xxx/yyy/somepackage
)

// @Success 200 {object} somepackage1.SomeStruct
// @Success 200 {object} somepackage2.SomeStruct
```

**Comment by bblbbl** [comment]
Use path like this `internal_api_dto.MobileObject`
